### DISCUSSION TITLE
Cloned VMs

### DISCUSSION THREAD

**Discussion by maw1146** [bb-Ricardo/netbox-sync#109] [General]
Hey!
I have an issue that I did not manage to fix.

Basically I have 2 vcenter cluster, in the past some VMs were copied from one to the other.
For these VM I have already updated the VMName and the MAC Address, but they still have the same uuid (seems not possible to change it in vCenter).

Would it be possible to "avoid" the check on the uuid and let netbox-sync to check only against name, IP and MAC Address?

Right now, when it sync the first Cluster, it generate the original VM in Netbox. But when it arrives to the second cluster it skips the copied VM.

Thanks

**Comment by bb-Ricardo** [comment]
Hi,

well, this should not be happening. Each source is handled as separate instance and a vm uuid from the first source should not be visible inside the second source. So I'm not sure why this is happening.

Are you sure this happens because of the same uuid? Can you use `-l DEBUG2` and see waht is exactly happening. Out here would be highly appreciated.

Thank you.

**Comment by maw1146** [comment]
Hi,
I have checked the log (using the DEBUG2) and found what is exactly doing.
NOTE: we have 2 VMWare cluster, some VLAN are configured on both cluster. The 2 Cluster are completely disconnected between them (a specific vCenter for each cluster).

Starting from an empty Netbox:
1) Sync Cluster 1 (with all his 5 VLANs)
2) Sync Cluster 2 -> when it encounters the VLAN that is present on the Cluster 1 it changes the VM that are in the VLAN (created on step 1) to Orphaned and generate the VM only for the new Cluster.

I'm not sure if I was clear on the explaination.

For security reason, I cannot send the out file here.

**Reply by bb-Ricardo** [reply]
Would you be able to send it to my email address? I won't disclose any information.

Or can you extract the relevant parts from the logs and change the sensitive details and post it here?

It seems an usecase or error which should be fixed.

And can you try to use the 'development' branch?

Thank you

**Comment by maw1146** [comment]
I'm pretty new on GitHub :)
I don't manage to find your mail address, where can I see it?

what do you mean by "can you try to use the 'development' branch"?

**Reply by bb-Ricardo** [reply]
On your command line you can do a `git log` and then you will see my email address. Don't worry about the development branch.

**Comment by bb-Ricardo** [comment]
Hi, I just reviewed the logs and I might have found the issue 😄 

You need to add all sources (vCenter) in ONE config file.
Example:

```ini
[common]
log_level = INFO

[netbox]
api_token = XYZ

# Requires a hostname or IP which points to your NetBox instance
host_fqdn = netbox.example.com

[source/vcenter-1]

enabled = True
host_fqdn = vcenter-1.services.domain.acme

[source/vcenter-2]

enabled = True
host_fqdn = vcenter-2.services.domain.acme
```

This should resolve your issues.

I will also describe this more explicitly in the README.

let me know if this helped.

**Reply by maw1146** [reply]
Hi Ricardo,
previously was like you mentioned (both vcenter in the same settings.ini file) and it was not working.
I have split them to test if something different would happen.

When they were on the same settings file, one of these machines that I wrote you, would not even be created (ignored completely).

if you wish, I can re-execute them using a single settings file and send you the DEBUG2 log.

**Reply by bb-Ricardo** [reply]
that would be great, Thank you.

**Reply by bb-Ricardo** [reply]
Thank you very much for the logs. You are absolutely correct about the uuid being already parsed. I had a misconception in my code which caused this bug.

I created issue #111 and pushed a fix to the branch `development`.  Let's continue our conversation over there.
